log_analytics.tf

```diff
@@ -81,3 +81,65 @@ resource "azurerm_log_analytics_solution" "main" {
     publisher = "Microsoft"
   }
 }
+
+locals {
+  dcr_location = try(coalesce(try(local.log_analytics_workspace.location, null), try(data.azurerm_log_analytics_workspace.main[0].location, null)), null)
+}
+
+resource "azurerm_monitor_data_collection_rule" "dcr" {
+  count               = (var.log_analytics_workspace_enabled && var.oms_agent_enabled) ? 1 : 0
+  name                = "MSCI-${local.dcr_location}-${azurerm_kubernetes_cluster.main.name}"
+  resource_group_name = data.azurerm_resource_group.main.name
+  location            = local.dcr_location
+  tags                = var.tags
+
+  destinations {
+    log_analytics {
+      workspace_resource_id = local.log_analytics_workspace.id
+      name                  = local.log_analytics_workspace.name
+    }
+  }
+
+  data_flow {
+    streams      = var.streams
+    destinations = [local.log_analytics_workspace.name]
+  }
+
+  data_flow {
+    streams      = ["Microsoft-Syslog"]
+    destinations = [local.log_analytics_workspace.name]
+  }
+
+  data_sources {
+    syslog {
+      streams        = ["Microsoft-Syslog"]
+      facility_names = var.syslog_facilities
+      log_levels     = var.syslog_levels
+      name           = "sysLogsDataSource"
+    }
+
+    extension {
+      streams        = var.streams
+      extension_name = "ContainerInsights"
+      extension_json = jsonencode({
+        "dataCollectionSettings" : {
+          "interval" : var.data_collection_settings.data_collection_interval,
+          "namespaceFilteringMode" : var.data_collection_settings.namespace_filtering_mode_for_data_collection,
+          "namespaces" : var.data_collection_settings.namespaces_for_data_collection,
+          "enableContainerLogV2" : var.data_collection_settings.container_log_v2_enabled,
+        }
+      })
+      name = "ContainerInsightsExtension"
+    }
+  }
+
+  description = "DCR for Azure Monitor Container Insights"
+}
+
+resource "azurerm_monitor_data_collection_rule_association" "dcra" {
+  count                   = (var.log_analytics_workspace_enabled && var.oms_agent_enabled) ? 1 : 0
```

Comment:
Would `count                   = (local.create_analytics_workspace && var.oms_agent_enabled) ? 1 : 0` be better?